src/time_travel/events_pool.py

```diff
@@ -8,23 +8,35 @@ def __init__(self):
         """Initialise an event pool."""
         self.future_events = {}
         
-    def add_future_event(self, timestamp, event):
+    def add_future_event(self, timestamp,
+                         event_descriptor,
+                         event_predicate=None):
         """Add an event to a given timestamp.
         
         - timestamp: time in seconds since the epoch.
         - event: any object (preferebly mock object) identifying the object
                  that a patched event waiting function will be wait on. 
         """
-        self.future_events.setdefault(timestamp, set()).add(event)
+        self.future_events.setdefault(timestamp, set()).add(
+            (event_descriptor, event_predicate))
         
-    def get_events(self):
+    def get_events(self, predicate=None):
         """Return all added events sorted by timestamp.
         
         The returned list is:
         [(timestamp1, set(events)), (timestamp2, set(events)), ....]
         where the timestamps are sorted.
         """
-        return sorted(self.future_events.items(), key=lambda x: x[0])
+        def filtered_set(events_set):
+            return set([event for event, event_type in
+                        events_set if predicate(event_type)])
+        
+        predicate = (lambda etype: True) if predicate is None else predicate
```

Comment:
It took me a while to get where this is used. You should move it to be before the private function definition.